projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js

```diff
@@ -6,39 +6,54 @@ import { useSelect } from '@wordpress/data';
 import { store as editorStore } from '@wordpress/editor';
 import { useEffect, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import { isModalSupportedByFlow } from './util';
 import './editor.scss';
 
 export const name = 'launchpad-save-modal';
 
 export const settings = {
 	render: function LaunchpadSaveModal() {
-		const isSaving = useSelect(
+		const isSavingSite = useSelect(
 			select => select( editorStore ).isSavingNonPostEntityChanges(),
 			[]
 		);
-		const prevIsSaving = usePrevious( isSaving );
+		const isSavingPost = useSelect( select => select( editorStore ).isSavingPost(), [] );
+
+		const prevIsSavingSite = usePrevious( isSavingSite );
+		const prevIsSavingPost = usePrevious( isSavingPost );
 		const [ isModalOpen, setIsModalOpen ] = useState( false );
 		const [ dontShowAgain, setDontShowAgain ] = useState( false );
 		const [ isChecked, setIsChecked ] = useState( false );
 
 		const { launchpadScreenOption, siteIntentOption } = window?.Jetpack_LaunchpadSaveModal || {};
-		const isInsideSiteEditor = document.getElementById( 'site-editor' ) !== null;
+		const isInsideSiteOrBlockEditor =
+			document.getElementById( 'site-editor' ) !== null ||
+			document.querySelector( '.block-editor' ) !== null;
 
 		const siteFragment = getSiteFragment();
 		const launchPadUrl = getRedirectUrl( 'wpcom-launchpad-setup-link-in-bio', {
 			query: `siteSlug=${ siteFragment }`,
 		} );
 
 		useEffect( () => {
-			if ( prevIsSaving === true && isSaving === false ) {
-				setIsModalOpen( true );
+			const siteEditorSaved = prevIsSavingSite === true && isSavingSite === false;
+			const postEditorSaved = prevIsSavingPost === true && isSavingPost === false;
+
+			if ( siteEditorSaved || postEditorSaved ) {
+				/* When the user publishes their first post in the post editor, for some reason it triggers the onRequestClose function which immediately closes the modal
+				 * It might have to do with the URL changing from /post-new.php to /post.php and potentially unmounting the component
+				 * A simple workaround I found to be working is to delay the rendering of the modal so the URL has already been changed by then
+				 */
+				setTimeout( () => {
```

Comment:
Not a fan of using async code to address timing problems unless necessary. The order of code execution isn't guaranteed and it can lead to unexpected results that aren't intuitive to debug.

Is it possible to detect some prop that changes during the first save that we can use to determine when to render the modal?

Reply:
I agree, I am not a fan of this either but I wasn't able to find anything so far. I think I can spend more time looking for sth in a follow up